src/Transfer/Target.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace Utopia\Transfer;
+
+use Utopia\Transfer\ResourceCache;
+
+abstract class Target
+{
+    /**
+     * Global Headers
+     *
+     * @var array
+     */
+    protected $headers = [
+        'Content-Type' => '',
+    ];
+
+    /**
+     * Resource Cache
+     *
+     * @var ResourceCache $resourceCache
+     */
+    public $resourceCache;
+
+    /**
+     * Endpoint
+     *
+     * @var string $endpoint
+     */
+    protected $endpoint = '';
+
+    /**
+     * Gets the name of the adapter.
+     *
+     * @return string
+     */
+    abstract public static function getName(): string;
+
+    /**
+     * Get Supported Resources
+     *
+     * @return array
+     */
+    abstract public function getSupportedResources(): array;
+
+    /**
+     * Register Transfer Cache
+     *
+     * @param ResourceCache &$cache
+     *
+     * @return void
+     */
+    public function registerTransferCache(ResourceCache &$cache): void
```

Comment:
This seems out of context, and the naming doesn't really fit any of out patterns.

Reply:
I've gone ahead and renamed it to registerCache, lmk what you think